rpc/src/lib.rs

```diff
@@ -218,203 +221,27 @@ pub fn new_http_client(host: String, rpc_auth: RpcAuthData) -> RpcClientResult<R
     HttpClientBuilder::default().set_http_middleware(middleware).build(host)
 }
 
-fn make_http_header_value(username: &str, password: &str) -> http::HeaderValue {
-    http::HeaderValue::from_str(&format!(
-        "Basic {}",
-        base64::engine::general_purpose::STANDARD.encode(format!("{username}:{password}"))
-    ))
-    .expect("Should not fail")
-}
-
-#[cfg(test)]
-mod tests {
-    use std::path::PathBuf;
-
-    use super::*;
-    use crypto::random::{
-        distributions::{Alphanumeric, DistString},
-        Rng,
-    };
-    use jsonrpsee::core::client::ClientT;
-    use jsonrpsee::rpc_params;
-    use rstest::rstest;
-    use test_utils::random::{make_seedable_rng, Seed};
-
-    #[rpc(server, namespace = "some_subsystem")]
-    pub trait SubsystemRpc {
-        #[method(name = "name")]
-        fn name(&self) -> RpcResult<String>;
-
-        #[method(name = "add")]
-        fn add(&self, a: u64, b: u64) -> RpcResult<u64>;
-    }
-
-    #[rpc(server, namespace = "example_server")]
-    trait RpcInfo {
-        #[method(name = "protocol_version")]
-        fn protocol_version(&self) -> RpcResult<String>;
-    }
-
-    struct RpcInfo;
-    impl RpcInfoServer for RpcInfo {
-        fn protocol_version(&self) -> RpcResult<String> {
-            Ok("version1".into())
-        }
-    }
+pub type RpcWsClient = jsonrpsee::ws_client::WsClient;
 
-    pub struct SubsystemRpcImpl;
+pub async fn new_ws_client(
+    host: impl AsRef<str>,
+    rpc_auth: RpcAuthData,
+) -> RpcClientResult<RpcWsClient> {
+    let headers = rpc_auth
+        .get_header()
+        .map(|header_val| http::HeaderMap::from_iter([(header::AUTHORIZATION, header_val)]))
+        .unwrap_or_default();
```

Comment:
Can you please explain why default? Why would this fail? Why not just expect on it?